stdlib/filename.ml

```diff
@@ -359,3 +359,13 @@ let open_temp_file ?(mode = [Open_text]) ?(perms = 0o600)
     with Sys_error _ as e ->
       if counter >= 1000 then raise e else try_name (counter + 1)
   in try_name 0
+
+let temp_dir ?(temp_dir = !current_temp_dir_name) prefix suffix =
+  let rec try_name counter =
+    let name = temp_file_name temp_dir prefix suffix in
+    try
+      Sys.mkdir name 0o700;
+      name
+    with Sys_error _ as e ->
+      if counter >= 1000 then raise e else try_name (counter + 1)
```

Comment:
The function looks good to me, but why did you pick 1000 retries? That seems like an awfully high number to spin on for avoiding a race: you are either in the presence of a malicious attacker at this point, or the server is under so much load that spinning won't help anyway. I notice that [musl uses a 100](https://github.com/bminor/musl/blob/master/src/temp/mkdtemp.c); I've not checked other mkdtemp implementations.

Reply:
Copy-paste from other examples in the same file.  Want me to submit a second PR to change them all?

Reply:
glibc uses [200k](https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/posix/tempname.c;h=60f8541085091934230e0adf96ce8a241531ca18;hb=c923cd8c496c7f253f327361a65c737233c7ebbd#l247).

Given that 1000 is already used in other functions, and that it's vaguely the geometric mean of the retries in musl and glibc, it seems good to keep as is.

(FWIW, the retries are there to deal with collisions, not races. If some process is leaking temp files, it creates potential collisions but there's no race)